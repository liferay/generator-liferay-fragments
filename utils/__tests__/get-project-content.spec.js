const getProjectContent = require('../get-project-content');
const path = require('path');

const SAMPLE_PROJECT_PATH = path.resolve(
  __dirname,
  'assets',
  'sample-project-content'
);

describe('utils/get-project-content', () => {
  it('requires a basePath', () => {
    expect(() => getProjectContent()).toThrow();
  });

  it('requires a valid package.json file', () => {
    expect(() => getProjectContent('')).toThrow();
  });

  it('matches a project structure', () => {
    const projectContent = getProjectContent(SAMPLE_PROJECT_PATH);

    projectContent.basePath = '';
    expect(projectContent).toMatchSnapshot();
  });
});
