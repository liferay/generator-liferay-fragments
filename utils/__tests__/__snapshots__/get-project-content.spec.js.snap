// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/get-project-content matches a project structure 1`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample component fragment name",
            "type": "component",
          },
          "slug": "sample-component-fragment",
        },
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration.json",
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample section fragment name",
            "type": "section",
          },
          "slug": "sample-section-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "project": Object {
    "name": "sample-project-content",
  },
}
`;

exports[`utils/get-project-content matches a project structure 2`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "{
  \\"fieldSets\\": []
}
",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "configurationPath": "configuration/configuration.json",
            "cssPath": "c/s/s/sample-fragment.css",
            "htmlPath": "h/t/m/l/sample-fragment.html",
            "jsPath": "j/s/sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "project": Object {
    "name": "sample-project-with-custom-paths",
  },
}
`;

exports[`utils/get-project-content matches a project structure 3`] = `
Object {
  "basePath": "",
  "collections": Array [
    Object {
      "fragmentCollectionId": "sample-collection",
      "fragmentCompositions": Array [],
      "fragments": Array [
        Object {
          "configuration": "",
          "css": ".fragment { color: inherit; }
",
          "html": "<div>Fragment</div>
",
          "js": "/* global fragmentElement */

fragmentElement.style.color = 'inherit';
",
          "metadata": Object {
            "cssPath": "sample-fragment.css",
            "htmlPath": "sample-fragment.html",
            "jsPath": "sample-fragment.js",
            "name": "Sample fragment name",
          },
          "slug": "sample-fragment",
        },
      ],
      "metadata": Object {
        "description": "Sample collection description",
        "name": "Sample collection name",
      },
      "slug": "sample-collection",
    },
  ],
  "project": Object {
    "name": "sample-project-without-fragment-configuration",
  },
}
`;
